include/logger/logger_lib.hpp

```diff
@@ -3,7 +3,62 @@
 
 #include "sink.hpp"
 
+/**
+ * @brief Macro FLAG_ON() extracts the log data such as date, time, line number,file name,function name from where it is being
+ called.Note to use this it has to be passed as function argument explicitly and works inside a function  only. * @param format_pattern pattern that will be used to format  the data 
+ */
+#define FLAG_ON() logger::extract_log_data(__DATE__, __TIME__, __FILE__, __LINE__, __FUNCTION__)
+
+ /**
+ * @brief  Macro FLAG_OFF() extracts the log date only date and time .
+ */
+#define FLAG_OFF() logger::extract_log_data(__DATE__, __TIME__)
```

Comment:
Why this  categorization ? you need all the Data the formatter  will decide which data to Keep or Logged

Reply:
This is default parameter as if user forgets or dont want to pass the flag it will only capture , date and time . 

Another reason to make this , Flag_ON()  contains __FUNCTION__ which is required to pass explicitly in the function then only it can capture function name .

Hence required to make two flags.

Reply:
Why cant we capture other parameters like the way we do date and time for FLAG_OFF